src/pamela/parser.clj

```diff
@@ -310,16 +310,30 @@
             (recur plant-opts (conj argvals v) (first more) (rest more))))))))
 
 (defn ir-fn [f & args]
+  (dbg-println :trace "IR-FN" f "ARGS" args)
   (loop [fn-opts {} body [] a (first args) more (rest args)]
+    (dbg-println :trace "  FN-OPTS" fn-opts "A" a)
     (if-not a
       (merge {:type f :body (if (empty? body) nil body)} fn-opts)
-      (if (and (vector? a) (#{:fn-opt :delay-opt} (first a)))
-        (recur (merge fn-opts (second a)) body (first more) (rest more))
-        (recur fn-opts (conj body a) (first more) (rest more))))))
+      (cond
+        (and (vector? a) (= :fn-opt (first a)))
+        (recur (merge fn-opts (second a)) body
+          (first more) (rest more))
+        (and (vector? a) (= :delay-opt (first a))
```

Comment:
A short comment would be useful here, distinguishing the assumptions behind this case and the next (`:delay-opt`)